src/Database/ODBC/Internal.hs

```diff
@@ -605,6 +607,40 @@ getData dbc stmt i col =
                    (fmap fromIntegral (odbc_TIME_STRUCT_hour datePtr)) <*>
                    (fmap fromIntegral (odbc_TIME_STRUCT_minute datePtr)) <*>
                    (fmap fromIntegral (odbc_TIME_STRUCT_second datePtr))))
+     | colType == sql_ss_timestampoffset ->
+       withCallocBytes
+         20
```

Comment:
Shouldn't this be 10 bytes? Why 20?

Reply:
I'm not sure, but I have a theory. I'd like to air it here; please let me know if this sounds completely bonkers.

I couldn't find anything in the documentation that states the size of the `TIMESTAMPOFFSET_STRUCT`, so I added the sizes of the constituent elements of the struct together. 3 SQLSMALLINT + 5 SQLUSMALLINT + 1 SQLUINTEGER. I used the conversion table in https://docs.microsoft.com/en-us/sql/odbc/reference/appendixes/c-data-types to translate those to C types, and from there I guessed at the size of each, based on https://en.wikipedia.org/wiki/C_data_types. I guessed that `short int` and `unsigned short int` would both be 2 bytes (the minimum size) and `unsigned long int` would be 4 bytes (again, the minimum size). That's 3 * 2 + 5 + 2 + 1 * 4 = 20.

I've noticed that https://docs.microsoft.com/en-us/sql/t-sql/data-types/datetimeoffset-transact-sql states that the _storage size_ of the `datetimeoffset` column type is 10 bytes, but I suppose that that's because it uses a more compact data format (something like nanoseconds since some zero date, plus the time zone offset...).

At a more pragmatic level, the code works when I allocate 20 bytes. If I try to allocate 10 bytes, it doesn't work:

```
  test\Main.hs:343:
  1) Database.ODBC.SQLServer, Conversion to SQL, QuickCheck roundtrip: HS=Datetimeoffset, SQL=datetimeoffset
       uncaught exception: ODBCException (UnsuccessfulReturnCode "getTypedData ty=SQLCTYPE (-2)" (-1) "[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range{") (after 1 test)
       Datetimeoffset {unDatetimeoffset = 1777-07-20 21:45:49.3091726 +0824}

  test\Main.hs:343:
  2) Database.ODBC.SQLServer, Conversion to SQL, QuickCheck roundtrip: HS=Maybe Datetimeoffset, SQL=datetimeoffset
       uncaught exception: ODBCException (UnsuccessfulReturnCode "getTypedData ty=SQLCTYPE (-2)" (-1) "[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range{") (after 1 test)
       Datetimeoffset {unDatetimeoffset = 1777-07-20 21:45:49.3091726 +0824}
```

Reply:
Fair enough. :+1:  If you can document that somewhere, then the code will be self-explanatory. Allocating 20 bytes isn't a big deal more than 10, I was just curious why it differed to the claim I saw in MSDN.

>   test\Main.hs:343:
  1) Database.ODBC.SQLServer, Conversion to SQL, QuickCheck roundtrip: HS=Datetimeoffset, SQL=datetimeoffset
       uncaught exception: ODBCException (UnsuccessfulReturnCode "getTypedData ty=SQLCTYPE (-2)" (-1) "[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range[Microsoft][ODBC Driver 17 for SQL Server]Numeric value out of range{") (after 1 test)
       Datetimeoffset {unDatetimeoffset = 1777-07-20 21:45:49.3091726 +0824}

Note to self, fix that error reporting. It's truncated intentionally, but out of laziness.

Reply:
Indeed, I've now added a comment in a new commit.

FWIW, I had in mind doing that all along, but I felt the need to run the above explanation by someone first, as I wasn't entirely sure that it was correct.